quickinput-sample/src/promptCommandWithHistory.ts

```diff
@@ -0,0 +1,101 @@
+import { window, Disposable, Memento } from 'vscode';
+import { QuickPickItem } from 'vscode';
+
+const HISTORY_KEY = 'HISTORY_KEY';
+
+
+/**
+ * A command prompt with history
+ *
+ */
+export const getPromptCommand = (memo: Memento) => {
+	const pickCommand = getPickCommand(memo);
+	// §TODO: cache promptCommand!
+	return async function promptCommand() {
+		const command = await pickCommand();
+		if (command) {
+			window.showInformationMessage(`You picked the following command: '${command}'`)
```

Comment:
Could you install the TSLint extension? There are a few semicolons missing and we generally use braces even when there is only a single statement inside the `if`.

Reply:
my bad, I haven't the tslint set up in mind